### PR TITLE
[CMake] Improvements related to `opengl` build configuration flag

### LINKED ISSUE #16250: ROOT doesn't build anymore with `opengl=ON` and `asimage=OFF`
See also this discussion here:
https://github.com/root-project/root/pull/15812#issuecomment-2252107168

The PR #15812 introduced a direct dependency of `graf3d/gl` on TASImage, which is only built if `asimage=ON`. It would be better if the PNG export would be done via the `TImage` plugin system, so there is no dependency at build time. Or if this can't be done, at least error out at configuration time if one attempts to build with this combination of flags.

Here is the culprit:
https://github.com/root-project/root/blob/master/graf3d/gl/src/TGLSdfFontMaker.cxx#L198

@osschar @linev 

See also:
  * https://root.cern.ch/doc/master/classTASImage.html
  * https://root.cern/doc/master/imgconv_8C.html

### PR DESCRIPTION
**Pull request by guitargeek**
1) Don't force user to build graf3d OpenGL features if `cocoa=ON`

2) Require asimage=ON in case of opengl=ON

More detail in the commit description. I have tested on macOS that the first commit does what it should.

Closes #16250.

### CODE REVIEW

**Comment by pcanal** [comment]
> Closes https://github.com/root-project/root/pull/16604.

What is the correct issue number?  This is the number of xrootd/ssl PR.

File: graf2d/cocoa/src/TGCocoa.mm [hunk 1]
```diff
@@ -13,11 +13,7 @@
 
 #include "TGCocoa.h"
 
-// We want to pickup ROOT's glew and not the system OpenGL coming from:
-// ROOTOpenGLView.h ->QuartzWindow.h->Cocoa.h
-// Allowing TU's which include the system GL and then glew (from TGLIncludes)
-// leads to gltypes.h redefinition errors.
-#include "TGLIncludes.h"
+#include <GL/glew.h>
```
Comment (guitargeek) [inline graf2d/cocoa/src/TGCocoa.mm:16]: The comment seemed out of date. The "TGLIncludes.h" header only included `<GL/glew.h>` anyway.

**Comment by guitargeek** [comment]
> > Closes #16604.
> 
> What is the correct issue number? This is the number of xrootd/ssl PR.

Thanks for pointing that out! I have updated the issue number in the PR and commit description.

**Comment by guitargeek** [comment]
By the way, I had to do the same as we do for `XRootD`: defining the target in SearchInstalledSoftware directory.

**Review by bellenot — APPROVED** [review]
LGTM, let's see what @couet think about it

**Comment by guitargeek** [comment]
The purpuse of the change is to fix the issue #16250 without forcing an unneeded implicit dependency of `cocoa` on `asimage`

**Pull request merged**